### ISSUE TITLE
[RyuJIT/arm32][JitStress=1] Assertion failed 'arg->OperGet() == GT_OBJ || arg->OperGet() == GT_FIELD_LIST'

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#14549]
From corefx x86_arm_altjit run, e.g. "@dotnet-bot test Windows_NT x86_arm_altjit Checked corefx_jitstress1".

Happens with `COMPlus_JitStress=1`.

Found in System.Runtime.Numerics.Tests:
```
E:\gh\corefx\bin\tests\System.Runtime.Numerics.Tests\netcoreapp-Windows_NT-Release-x86>RunTests.cmd E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\
Using E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\ as the test runtime folder.
Executing in E:\gh\corefx\bin\tests\System.Runtime.Numerics.Tests\netcoreapp-Windows_NT-Release-x86\
Running tests... Start time: 13:52:50.02

E:\gh\corefx\bin\tests\System.Runtime.Numerics.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJit=*
E:\gh\corefx\bin\tests\System.Runtime.Numerics.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitNgen=*
E:\gh\corefx\bin\tests\System.Runtime.Numerics.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitName=protononjit.dll
E:\gh\corefx\bin\tests\System.Runtime.Numerics.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_NoGuiOnAssert=1
E:\gh\corefx\bin\tests\System.Runtime.Numerics.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_ContinueOnAssert=0
E:\gh\corefx\bin\tests\System.Runtime.Numerics.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_AltJitAssertOnNYI=1
E:\gh\corefx\bin\tests\System.Runtime.Numerics.Tests\netcoreapp-Windows_NT-Release-x86>set COMPlus_JitStress=1

E:\gh\corefx\bin\tests\System.Runtime.Numerics.Tests\netcoreapp-Windows_NT-Release-x86>call E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\\dotnet.exe xunit.console.netcore.exe System.Runtime.Numerics.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonwindowstests  -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Runtime.Numerics.Tests
Discovered:  System.Runtime.Numerics.Tests
Starting:    System.Runtime.Numerics.Tests

Assert failure(PID 16056 [0x00003eb8], Thread: 3048 [0x0be8]): Assertion failed 'arg->OperGet() == GT_OBJ || arg->OperGet() == GT_FIELD_LIST' in 'System.Numerics.Tests.ComplexTests:Conjugate(double,double)' (IL size 57)

    File: e:\gh\coreclr2\src\jit\lower.cpp Line: 853
    Image: E:\gh\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x86\dotnet.exe
```

Full repro:
```
dotnet/coreclr:
E:\gh\coreclr2> build.cmd x86 checked

dotnet/corefx:
E:\gh\corefx> build.cmd -Release -os:Windows_NT -buildArch:x86 -- /p:CoreCLROverridePath=E:\gh\coreclr2\bin\Product\Windows_NT.x86.Checked
E:\gh\corefx> build-tests.cmd -Release -os:Windows_NT -buildArch:x86 -- /p:WithoutCategories=IgnoreForCI /p:PreExecutionTestScript=e:\bugs\SetStressModes.bat

With e:\bugs\SetStressModes.bat:
set COMPlus_AltJit=*
set COMPlus_AltJitNgen=*
set COMPlus_AltJitName=protononjit.dll
set COMPlus_NoGuiOnAssert=1
set COMPlus_ContinueOnAssert=0
set COMPlus_AltJitAssertOnNYI=1
set COMPlus_JitStress=1
```

**Comment by BruceForstall** [comment]
@dotnet/arm32-contrib

**Comment by YongseopKim** [comment]
@BruceForstall `set COMPlus_AltJitName=protononjit.dll` always should be set ? I have no idea where I can get it. 
Except for that,
```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  ./run_fxutc_ryujit_arm.sh
LOG: FX_RYUJIT_10191259.log
export COMPlus_AltJit="*"
export COMPlus_AltJit="*"
export COMPlus_AltJitNgen="*"
#export COMPlus_AltJitName="protononjit.dll"
export COMPlus_NoGuiOnAssert=1
export COMPlus_ContinueOnAssert=0
export COMPlus_AltJitAssertOnNYI=1
export COMPlus_JitStress=1
/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/run-test.sh --runtime /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/overlay --corefx-tests /home/dragon/dotnet/arm32-dotnet/tmp/coreoverl
ay-ARM-CHE-REL-10181724/tests --os Linux --arch x64 --configurationGroup Release --test-dir-file ./tests_fx.txt

Running tests in /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/tests/System.Runtime.Numerics.Tests/netcoreapp-Linux-Release-x64
./RunTests.sh /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/overlay

Using as the test runtime folder.
Running tests... Start time: 12:59:46
Commands:
export XUNIT_PERFORMANCE_MIN_ITERATION=1
export XUNIT_PERFORMANCE_MAX_ITERATION=1
chmod +x /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/overlay/dotnet
/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/overlay/dotnet xunit.console.netcore.exe System.Runtime.Numerics.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxt
ests  -notrait category=failing
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/tests/System.Runtime.Numerics.Tests/netcoreapp-Linux-Release-x64 ~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/tests/System.Runtime.Numerics.Tests/netcoreapp-L
inux-Release-x64
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Runtime.Numerics.Tests
Discovered:  System.Runtime.Numerics.Tests
Starting:    System.Runtime.Numerics.Tests

Assert failure(PID 20308 [0x00004f54], Thread: 20308 [0x4f54]): Assertion failed '!BitVecOps::IsMember(&nodesVecTraits, uniqueNodes, gtTreeID)' in 'System.Reflection.Emit.DynamicMethod:GetDynamicMethodsModule():ref' (IL size 199)

    File: /home/dragon/dotnet/arm32-dotnet/coreclr/src/jit/flowgraph.cpp Line: 21782
    Image: /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/overlay/dotnet


Assert failure(PID 20308 [0x00004f54], Thread: 20308 [0x4f54]): Assertion failed '!BitVecOps::IsMember(&nodesVecTraits, uniqueNodes, gtTreeID)' in 'System.Reflection.Emit.DynamicMethod:GetDynamicMethodsModule():ref' (IL size 199)

    File: /home/dragon/dotnet/arm32-dotnet/coreclr/src/jit/flowgraph.cpp Line: 21782
    Image: /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/overlay/dotnet

./RunTests.sh: line 86: 20308 Aborted                 (core dumped) $RUNTIME_PATH/dotnet xunit.console.netcore.exe System.Runtime.Numerics.Tests.dll -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlin
uxtests -notrait category=failing
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10181724/tests/System.Runtime.Numerics.Tests/netcoreapp-Linux-Release-x64
Finished running tests. End time=13:00:22. Return value was 134
Unable to find executable corerun
error: One or more tests failed while running tests from ''.  Exit code 134.
1 test(s) failed
```

**Comment by YongseopKim** [comment]
```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  cat tmp/coreoverlay-ARM-CHE-REL-10181724/COMMITS.CLR 
e869941a7364ef128a50ecdfce0a9a0e317d13c2 format spmi sources. (#14545)
a194ac80eb0e97e822b9eb35c24c59c3c24d7693 Merge pull request #14553 from briansull/fix-warn
882b1d64a179f2d04dc66c8845b723f06c30bdea Merge pull request #14555 from wtgodbe/AzureBlobSymPkg
de4559f6dbaa7fd732f6529e93b0aebde0fb0ea9 Use C#-friendly path for indexing symbol packages
d183b9f0cb9235440902eb6f80841830d96f5775 Fix for x86 desktop build Fix warning C4389: '==' : signed/unsigned mismatch
4fa06e6095af6473d2ec6c855ebfb8d77017d91d Merge pull request #14541 from stephentoub/taskbeginwait
2f01c285d1a2d01cf9d8810e557961872c1eb46d GetEnvironmentVariable: Avoid StringBuilder marshaling overhead (#14502)
2cd4ca4cdd8b71563a58835850aa67de5906821c CLRLifoSemaphore cleanup (#14535)
0526339a302da4918699901d5fc403044374651b Avoid TaskWaitBegin/End events in Task.Wait on already completed task
b5616f0013ccba23c6a3e6facbb09545e710a5f1 Merge pull request #14493 from alpencolt/armel-cross-build
```

```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  cat tmp/coreoverlay-ARM-CHE-REL-10181724/COMMITS.FX 
5c8af507d63e94a49ab4b4a81a45510a50146faf Adding ValidateTypeIsBlittable SpanHelper method and rename constant (#24626)
b28112eb892283f20e7e422a52da0a55ca4881d5 Rewrite BackgroundWorker test to be non blocking (#24693)
ae86eca620025087b576d092e67acde7760d4baa Deleted unused string resources. (#24697)
a1e7009bd3c94c3999bfd471a08e5d647c571e52 Add null check when setting Packaging property (#24673)
0ad10b8bf74a4ce2fe452aa23801b3722237560d Update CoreSetup, Standard to preview1-25817-01, preview1-25817-02, respectively (#24702)
5ef5f17b5606377415d5c171d6613c49c59a1b55 Set.TryGetValue not-found behavior/docs inconsistent (#24699)
79089260b0bf00e99ba9dd7e736c7b2d776fed0a Deleted dead code from System.IO.FileSystem. (#24701)
3b867880d9992f305757db85a9b32044f9773863 Fix for GraphicsPathIterator.CopyData throws EntrypointNotFoundException when index is out of range. (#24656)
e3977683cf78b7b1896a11e9de7df6a62408a1a6 Remove methods marked as internal from reference assembly. (#24700)
db596d27932201ec4936a7c9baa6aa6ad3f9e796 Update XunitNetcoreExtensionsVersion to 1.0.1-prerelease-02116-01 (#24687)
```

**Comment by BruceForstall** [comment]
> @BruceForstall set COMPlus_AltJitName=protononjit.dll always should be set ? I have no idea where I can get it.

On Windows, protononjit.dll in an x86 build is an arm-targeting cross JIT. On Linux, use protojit.dll (which is the default for `COMPlus_AltJitName`)

**Comment by BruceForstall** [comment]
This is a recent regression: https://github.com/dotnet/coreclr/issues/14567

**Comment by YongseopKim** [comment]
With d780f9a1327344bfc8e06737497b703cdbfba804, the tc passes on linux/arm32.

```
 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  ./run_fxutc_ryujit_arm.sh 
LOG: FX_RYUJIT_10251024.log
export COMPlus_AltJit="*"
export COMPlus_AltJitNgen="*"
export COMPlus_ContinueOnAssert=0
export COMPlus_AltJitAssertOnNYI=1
export COMPlus_JitStress=1
/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/run-test.sh --runtime /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/overlay --corefx-tests /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/tests --os Linux --arch x64 --configurationGroup Release --test-dir-file ./tests_fx2.txt

Running tests in /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/tests/System.Runtime.Numerics.Tests/netcoreapp-Linux-Release-x64
./RunTests.sh /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/overlay

Using as the test runtime folder.
Running tests... Start time: 10:24:40
Commands:
export XUNIT_PERFORMANCE_MIN_ITERATION=1
export XUNIT_PERFORMANCE_MAX_ITERATION=1
chmod +x /home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/overlay/dotnet
/home/dragon/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/overlay/dotnet xunit.console.netcore.exe System.Runtime.Numerics.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests  -notrait category=failing
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/tests/System.Runtime.Numerics.Tests/netcoreapp-Linux-Release-x64 ~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/tests/System.Runtime.Numerics.Tests/netcoreapp-Linux-Release-x64
xUnit.net console test runner (32-bit .NET Core)
Copyright (C) 2014 Outercurve Foundation.

Discovering: System.Runtime.Numerics.Tests
Discovered:  System.Runtime.Numerics.Tests
Starting:    System.Runtime.Numerics.Tests
Finished:    System.Runtime.Numerics.Tests

=== TEST EXECUTION SUMMARY ===
   System.Runtime.Numerics.Tests  Total: 1398, Errors: 0, Failed: 0, Skipped: 0, Time: 1228.506s
~/dotnet/arm32-dotnet/tmp/coreoverlay-ARM-CHE-REL-10251012/tests/System.Runtime.Numerics.Tests/netcoreapp-Linux-Release-x64
Finished running tests. End time=10:45:39. Return value was 0
All tests passed.

 dragon@dragon-rasp3  ~/dotnet/arm32-dotnet  cat ./tmp/coreoverlay-ARM-CHE-REL-10251012/COMMITS.CLR 
d780f9a1327344bfc8e06737497b703cdbfba804 fix lvaStressLclFldCB (#14676)
f324495d90ebe7cba7e10261a53cd82383bb317e Merge pull request #14658 from mikedn/legacy-reg-track
f09dd7ca32ff62dc25db9a0de0e6400183cddb58 Merge pull request #14619 from briansull/emitter-cleanup
dbfd10c94055fb5af823b4a7e30774641afcb3b2 Delete dead code (#14673)
1a0e2ef9c559af14acea429aecc591e1550b57db Merge pull request #14662 from wtgodbe/FeedTaskPackage
b813321f918b91142e95bc6b74d524664ea703ba Use new feed tasks package & auto-update version
4419496eb27ccc705c19c79ab4df714c06f507c3 Ifdef out legacy RegTracker code
b5a2484a9c1dbc751f77fe1934149c78d7f0ffc1 Cleanup unused emitter arguments Removed unused idClsCookie from struct instrDescDebugInfo Cleaned up several ifdefs
15e1472aab7ed42dafde0331ed6cece465a0763d Update BuildTools, CoreClr, CoreFx to prerelease-02120-09, preview1-25822-02, preview1-25824-01, respectively (#14565)
8c74b3bcd691d842a891bc55909a18a8b1e53200 Search/replace _DEBUG => DEBUG
```

**Comment by BruceForstall** [comment]
I see a failure with JitStress=2 with altjit on:
```
JIT\SIMD\Matrix4x4_r\Matrix4x4_r.cmd
```
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x86_arm_altjit_checked_windows_nt_jitstress2_prtest/6/consoleText

**Comment by sandreenko** [comment]
Fixed in #14753
